app/post_test.go

```diff
@@ -346,3 +349,57 @@ func TestMakeOpenGraphURLsAbsolute(t *testing.T) {
 		})
 	}
 }
+
+func TestMaxPostSize(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		Description         string
+		StoreMaxPostSize    int32
+		ExpectedMaxPostSize int
+		ExpectedError       *model.AppError
+	}{
+		{
+			"error fetching max post size",
+			0,
+			model.POST_MESSAGE_MAX_RUNES_V1,
+			model.NewAppError("TestMaxPostSize", "this is an error", nil, "", http.StatusBadRequest),
+		},
+		{
+			"4000 rune limit",
+			4000,
+			4000,
+			nil,
+		},
+		{
+			"16383 rune limit",
+			16383,
+			16383,
+			nil,
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.Description, func(t *testing.T) {
```

Comment:
Maybe you meant to add a `t.Parallel()` here? Otherwise, the `testCase := testCase` above doesn't really do anything.

Reply:
You are correct on both counts :)